decidim-core/app/controllers/decidim/account_controller.rb

```diff
@@ -4,23 +4,42 @@
 module Decidim
   # The controller to handle the user's account page.
   class AccountController < ApplicationController
-    helper_method :authorizations, :handlers
-    authorize_resource :user_account, class: false
+    helper_method :authorizations
+    include Decidim::UserProfile
 
-    private
+    def show
+      authorize! :show, current_user
+      @account = form(AccountForm).from_model(current_user)
+    end
+
+    def update
+      authorize! :update, current_user
+      @account = form(AccountForm).from_params(params)
+
+      UpdateAccount.call(current_user, @account) do
+        on(:ok) do |account|
+          flash.now[:notice] = if current_user.unconfirmed_email.present?
```

Comment:
Maybe you could broadcast something from the command so the controller doesn't know the internals of this.